src/overlays/actors/ovl_Obj_Hsblock/z_obj_hsblock.c

```diff
@@ -27,27 +31,142 @@ const ActorInit Obj_Hsblock_InitVars = {
+                            this->dyna.actor.posRot.pos.y, this->dyna.actor.posRot.pos.z, this->dyna.actor.posRot.rot.x,
+                            this->dyna.actor.posRot.rot.y, this->dyna.actor.posRot.rot.z, 1);
+    }
+}
+
+void ObjHsblock_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHsblock* this = THIS;
+
+    func_80B93B68(this, globalCtx, D_80B940DC[thisx->params & 3], 0);
+    Actor_ProcessInitChain(thisx, sInitChain);
+    func_80B93BF0(this, globalCtx);
+
+    switch (thisx->params & 3) {
+        case 0:
+        case 2:
+            func_80B93D90(this);
+            break;
+        case 1:
+            if (Flags_GetSwitch(globalCtx, (thisx->params >> 8) & 0x3F)) {
+                func_80B93D90(this);
+            } else {
+                func_80B93DB0(this);
+            }
+    }
+
+    mREG(13) = 255;
+    mREG(14) = 255;
+    mREG(15) = 255;
+}
+
+void ObjHsblock_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHsblock* this = THIS;
+
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void func_80B93D90(ObjHsblock* this) {
+    ObjHsblock_SetupAction(this, NULL);
+}
+
+void func_80B93DB0(ObjHsblock* this) {
+    this->dyna.actor.flags |= 0x10;
+    this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y - 105.0f;
+    ObjHsblock_SetupAction(this, func_80B93DF4);
+}
+
+void func_80B93DF4(ObjHsblock* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F)) {
+        func_80B93E38(this);
+    }
+}
+
+void func_80B93E38(ObjHsblock* this) {
+    ObjHsblock_SetupAction(this, func_80B93E5C);
+}
+
+void func_80B93E5C(ObjHsblock* this, GlobalContext* globalCtx) {
+    Math_SmoothScaleMaxMinF(&this->dyna.actor.velocity.y, 16.0f, 0.1f, 0.8f, 0.0f);
+    if (fabsf(Math_SmoothScaleMaxMinF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y, 0.3f,
+                                      this->dyna.actor.velocity.y, 0.3f)) < 0.001f) {
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y;
+        func_80B93D90(this);
+        this->dyna.actor.flags &= ~0x10;
+    }
+}
+
+void ObjHsblock_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHsblock* this = THIS;
+
+    if (this->actionFunc != NULL) {
+        this->actionFunc(this, globalCtx);
+    }
+    Actor_SetHeight(thisx, D_80B940C0[thisx->params & 3]);
+}
+
+void ObjHsblock_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    Color_RGB8* color;
+    Color_RGB8 defaultColor;
+
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    Gfx* dispRefs[4];
+
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_obj_hsblock.c", 365);
+    func_80093D18(globalCtx->state.gfxCtx);
+
+    gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_obj_hsblock.c", 369),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    if (globalCtx->sceneNum == SCENE_HIDAN) {
+        color = &sHidanColor;
```

Comment:
I personally like to avoid using their bad names when possible. theyre going to change at some point for the enum and stuff but for variables names id just call it sFireTempleColor now